File: Python/arranging-coins.py
class Solution:
    def arrangeCoins(self, n):
        """
        :type n: int
        :rtype: int
        """
        left, right = 1, n
        while left <= right:
            mid = left + (right - left) // 2
            if 2 * n < mid * (mid+1):
                right = mid - 1
            else:
                left = mid + 1
        return left - 1

File: Swift/arranging-coins.swift
class Solution {
    func arrangeCoins(_ n: Int) -> Int {
        var left = 1
        var right = n

        while left <= right {
            let mid = left + (right - left) / 2
            if n * 2 < mid * (mid + 1) {
                right = mid - 1
            } else {
                left = mid + 1
            }
        }

        return left - 1
    }
}
